exporters/stats/stackdriver/src/main/java/io/opencensus/exporter/stats/stackdriver/StackdriverExportUtils.java

```diff
@@ -374,54 +377,77 @@ static MonitoredResource getDefaultResource() {
-    switch (autoDetectedResource.getResourceType()) {
-      case GCP_GCE_INSTANCE:
-        GcpGceInstanceMonitoredResource gcpGceInstanceMonitoredResource =
-            (GcpGceInstanceMonitoredResource) autoDetectedResource;
-        builder.putLabels(PROJECT_ID_LABEL_KEY, gcpGceInstanceMonitoredResource.getAccount());
-        builder.putLabels("instance_id", gcpGceInstanceMonitoredResource.getInstanceId());
-        builder.putLabels("zone", gcpGceInstanceMonitoredResource.getZone());
-        return;
-      case GCP_GKE_CONTAINER:
-        GcpGkeContainerMonitoredResource gcpGkeContainerMonitoredResource =
-            (GcpGkeContainerMonitoredResource) autoDetectedResource;
-        builder.putLabels(PROJECT_ID_LABEL_KEY, gcpGkeContainerMonitoredResource.getAccount());
-        builder.putLabels("cluster_name", gcpGkeContainerMonitoredResource.getClusterName());
-        builder.putLabels("container_name", gcpGkeContainerMonitoredResource.getContainerName());
-        builder.putLabels("namespace_name", gcpGkeContainerMonitoredResource.getNamespaceId());
-        builder.putLabels("pod_name", gcpGkeContainerMonitoredResource.getPodId());
-        builder.putLabels("location", gcpGkeContainerMonitoredResource.getZone());
-        return;
-      case AWS_EC2_INSTANCE:
-        AwsEc2InstanceMonitoredResource awsEc2InstanceMonitoredResource =
-            (AwsEc2InstanceMonitoredResource) autoDetectedResource;
-        builder.putLabels("aws_account", awsEc2InstanceMonitoredResource.getAccount());
-        builder.putLabels("instance_id", awsEc2InstanceMonitoredResource.getInstanceId());
-        builder.putLabels("region", "aws:" + awsEc2InstanceMonitoredResource.getRegion());
+    Map<String, String> mappings = null;
+    switch (type) {
+      case ResourceKeyConstants.GCP_GCE_INSTANCE_TYPE:
+        builder.setType(GCP_GCE_INSTANCE);
+        mappings = GCP_RESOURCE_MAPPING;
+        break;
+      case ResourceKeyConstants.GCP_GKE_INSTANCE_TYPE:
+        builder.setType(GCP_GKE_CONTAINER);
+        mappings = GKE_RESOURCE_MAPPING;
+        break;
+      case ResourceKeyConstants.AWS_EC2_INSTANCE_TYPE:
+        builder.setType(AWS_EC2_INSTANCE);
+        mappings = AWS_RESOURCE_MAPPING;
+        break;
+      default:
+        builder.setType(GLOBAL);
         return;
     }
+
+    if (mappings != null) {
+      Map<String, String> resLabels = autoDetectedResource.getLabels();
+      for (Map.Entry<String, String> entry : mappings.entrySet()) {
+        if (resLabels.containsKey(entry.getValue())) {
+          builder.putLabels(entry.getKey(), resLabels.get(entry.getValue()));
+        }
+      }
+      return;
+    }
+
     throw new IllegalArgumentException("Unknown subclass of MonitoredResource.");
   }
 
+  private static Map<String, String> getGcpResourceLabelsMappings() {
+    Map<String, String> resourceLabels = new LinkedHashMap<String, String>();
```

Comment:
nit: This should be an `UnmodifiableMap` if you want to make it a constant. Same for all maps below.

Reply:
done